### PR TITLE
[CIVP-19741] Add _check_column_types method

### PR DESCRIPTION
**Pull request by zacharydfriedlander**
We can use this method to determine when there is a mismatch between column types or counts. Currently it raises a `CivisImportError`, but one we have completed other work on the Civis Platform to expose `loosen_types` in the `imports/files/csv`API we can refactor as we'd like.

### CODE REVIEW

File: civis/io/_tables.py [hunk 1]
```diff
@@ -1335,3 +1329,58 @@ def _process_cleaning_results(cleaning_futures, client, headers,
...
+                                   len(file_columns))
+                               )
+
+    error_msgs = []
+    for idx, (tcol, fcol) in enumerate(zip(table_columns, file_columns)):
+        # for the purposes of type checking, we care only that the types
+        # share a base type (e.g. INT, VARCHAR, DECIMAl) rather than that
+        # they have the same precision and length
+        # (e.g VARCHAR(42), DECIMAL(8, 10))
+        tcol_base_type = tcol['sql_type'].split('(', 2)[0]
```
Comment (jkulzick) [inline civis/io/_tables.py]: nit, but I think `.split('(', 1)` will suffice for both of these. It appears that the number of elements returned is actually `maxsplit + 1`. See [here](https://docs.python.org/3/library/stdtypes.html#str.split).

**Review by jkulzick — CHANGES_REQUESTED** [review]
One minor comment

File: civis/io/_tables.py (hunk 1 above)
Comment (zacharydfriedlander) [reply civis/io/_tables.py]: You are correct - I was misremembering the parameter's meaning. I thought it was the maximum number of pieces into which to break. Reduced to 1.

**Review by jkulzick — APPROVED** [review]
LGTM %Travis

**Pull request merged**